### PR TITLE
Allow viewconf to override min dims with `minHeight` or `minWidth` props

### PR DESCRIPTION
**Pull request by alexpreynolds**
## Description

> What was changed in this pull request?

When a viewconf is generated programmatically, a track height or width may be less than preset defaults. This PR modifies that check to look for a `minHeight` or `minWidth` parameter set in the viewconf track, and use that to apply new bounds.

> Why is it necessary?

Helps resolve discussion on Slack (https://higlass.slack.com/archives/CCDG12ZKP/p1568182133018000)

## Checklist

- [ ] Set proper GitHub labels (e.g. v1.6+, ignore if you don't know)
- [ ] Unit tests added or updated
- [ ] Documentation added or updated
- [ ] Example(s) added or updated
- [ ] Update schema.json if there are changes to the viewconf JSON structure format
- [ ] Screenshot for visual changes (e.g. new tracks or UI changes)
- [x] Updated CHANGELOG.md

### CODE REVIEW

**Comment by pkerpedjiev** [comment]
Thanks Alex! I think this looks  reasonable. Would you mind adding a test?

For testing locally, you can go into `karma.conf.js` and comment out all the irrelevant tests and run `npm run test-watch` to start yours.

**Comment by pkerpedjiev** [comment]
Oh, and would min updating the docs as well? https://github.com/higlass/higlass/blob/develop/docs/view_config.rst

**Comment by alexpreynolds** [comment]
I haven't worked on the higlass client itself, before, as well as how testing is done. I'm sorry about not doing this the correct way. I'll apply some fixes and resolve these issues.

**Comment by pkerpedjiev-zymergen** [comment]
> I haven't worked on the higlass client itself, before, as well as how testing is done. I'm sorry about not doing this the correct way. I'll apply some fixes and resolve these issues.

No worries at all. Really appreciate your contribution :-)

File: karma.conf.js [hunk 1]
```diff
@@ -98,13 +98,17 @@ module.exports = (config) => {
     colors: true,
     logLevel: config.LOG_DEBUG,
     autoWatch: true,
-    browsers: ['Chrome'],
+    browsers: ['Chrome', 'HeadlessChrome', 'MyHeadlessChrome'],
```
[the remark this answers is not in the document]
Comment (alexpreynolds) [reply karma.conf.js]: No difference, probably, and I'll likely excise the `MyHeadlessChrome` launcher. Mainly I copied-and-pasted stuff from Google HQ that was claimed to work (and which does) to minimize things to troubleshoot: https://developers.google.com/web/updates/2017/06/headless-karma-mocha-chai

[the remark this answers is not in the document]
Comment (alexpreynolds) [reply karma.conf.js]: The use of a name different from an internally-reserved `base` browser name (`ChromeHeadless`) appears to be a requirement, i.e., I can use `HeadlessChrome` but not `ChromeHeadless` in the browsers list. I will remove `MyHeadlessChrome`, which seems redundant.

File: test/HorizontalMultivecTests.js [hunk 2]
```diff
@@ -265,3 +335,143 @@ const viewConf1 = {
     locksDict: {}
   }
 };
+
+// minHeight view test
+const viewConf2 = {
```
[the remark this answers is not in the document]
Comment (alexpreynolds) [reply test/HorizontalMultivecTests.js]: I placed the viewconf in `test/view-configs-more`. Would that be acceptable?

**Comment by alexpreynolds** [comment]
Travis failed with the following error:

```
test/view-configs-more/horizontalMultivecWithSmallerDimensions.json invalid
data.views[0].layout should NOT have additional properties
```

Those additional props are `minHeight` and `minWidth`. 

Is their addition to the schema required? If so, I'd ask for guidance on what to do (what file to edit and so forth). Sorry!

**Comment by pkerpedjiev** [comment]
Oh, I think you just have to remove the `"moved"` and `"static"` fields from the `"layout"` section of the viewconf.

Here's the schema section: https://github.com/higlass/higlass/blob/26917a8a96da81a7262296fb2ef237d8b79504dc/app/schema.json#L194

I don't believe you need to add anything extra for the new options.

**Comment by pkerpedjiev** [comment]
OMG, I'm sorry, I pointed you in the wrong direction again. It looks like you have `minWidth` and `minHeight` under the top level `track` definition. I think they would be more appropriate under `options`. Then you won't have issues with the schema.

**Comment by pkerpedjiev** [comment]
You know what? If it's not urgent maybe I can work on this a little bit and try to get it merged this weekend? I'd like to move all the `minWidth` and `minHeight`  fields in `tracks-info.js` to the track's `defaultOptions`.

**Comment by alexpreynolds** [comment]
No urgency, at all! Just striking while the iron is hot.

**Comment by pkerpedjiev** [comment]
> striking while the iron is hot.

Awesome! 💪

I wasn't able to push to your branch so I duplicated your PR here: https://github.com/higlass/higlass/pull/776 It has your changes and a few more. If you have a moment, could you try it out to see if it works for you?

**Comment by alexpreynolds** [comment]
New PR seems to work fine! https://github.com/higlass/higlass/pull/776#issuecomment-531859547

**Pull request closed without merging**